src/develop/masks/brush.c

```diff
@@ -941,8 +941,12 @@ static void dt_brush_get_distance(float x, int y, float as, dt_masks_form_gui_t
 static int dt_brush_get_points_border(dt_develop_t *dev, dt_masks_form_t *form, float **points,
                                       int *points_count, float **border, int *border_count, int source)
 {
-  return _brush_get_points_border(dev, form, 999999, dev->preview_pipe, points, points_count, border,
-                                  border_count, NULL, NULL, source);
+  dt_pthread_mutex_lock(&dev->preview_pipe_mutex);
```

Comment:
that one is fine, i don't think anything other than this mutex will be locked when the input changes. unfortunately this mutex will also be locked for the whole process()ing of the preview pipe. did you notice any speed impact from this?